### PR TITLE
Allow injection on iframes

### LINKED ISSUE #1154: no web3 injected into iframes
This appears to be the same issue that the plugin had way back when:
https://github.com/MetaMask/metamask-extension/issues/1223

I can see web3 in the parent frame but any child iframes do not have `window.web3` available

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes #1154  along with a few bugs / perf improvements  in the browser

### CODE REVIEW

File: app/core/BackgroundBridge.js [hunk 1]
```diff
@@ -7,13 +7,27 @@ export class BackgroundBridge {
...
 
-	_postMessageToProvider(message) {
+	_postMessageToProvider(message, origin) {
 		const current = this._webview.current;
-		current &&
-			current.injectJavaScript(`window.ethereum && window.ethereum._onMessage(${JSON.stringify(message)})`);
+		// Loop through the iframes first
+		// If the source doesn't match any
+		// send the message to the main window
+		const js = `(function () {
```
Comment (estebanmino) [inline app/core/BackgroundBridge.js]: move it to . https://github.com/MetaMask/metamask-mobile/blob/develop/app/util/browserScripts.js ?

**Review by estebanmino — APPROVED** [review]
🔥

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good on both OS's, QA Passed 👍

**Review by M66rr — COMMENTED** [review]
Marged✅

**Pull request merged**